moto/rds/models.py

```diff
@@ -1400,6 +1421,20 @@ def physical_resource_id(self) -> Optional[str]:
     def latest_restorable_time(self) -> datetime:
         return utcnow()
 
+    @property
+    def associated_roles(self) -> list[dict[str, Any]]:  # type: ignore[misc]
```

Comment:
Is it possible to just return `self._associated_roles` here? Our response serializer should automatically parse the `AssociatedRole`-class and convert it into a dict, we shouldn't have to manually handle this.

If this does not work, then there is something wrong with our serializer, and we can stick with the current implementation - but returning the `list[AssociatedRole]` directly _should_ work :slightly_smiling_face: